components/dashboard/src/teams/OrgSettingsPage.tsx

```diff
@@ -17,17 +21,43 @@ export interface OrgSettingsPageProps {
 
 export function OrgSettingsPage({ children }: OrgSettingsPageProps) {
     const team = useCurrentTeam();
-    const { data: teamBillingMode } = useOrgBillingMode();
-    const { oidcServiceEnabled, orgGitAuthProviders } = useContext(FeatureFlagContext);
-    const menu = getTeamSettingsMenu({
-        team,
-        billingMode: teamBillingMode,
-        ssoEnabled: oidcServiceEnabled,
-        orgGitAuthProviders,
-    });
+    const { data: teamBillingMode, isLoading: isBillingModeLoading } = useOrgBillingMode();
+    const { isOwner, isLoading: isMemberInfoLoading } = useCurrentOrgMember();
+    const { oidcServiceEnabled, orgGitAuthProviders } = useFeatureFlags();
+    const isLoading = isBillingModeLoading || isMemberInfoLoading;
+
+    const menu = useMemo(
+        () =>
+            getTeamSettingsMenu({
+                team,
+                billingMode: teamBillingMode,
+                ssoEnabled: oidcServiceEnabled,
+                orgGitAuthProviders,
+            }),
+        [oidcServiceEnabled, orgGitAuthProviders, team, teamBillingMode],
+    );
+
+    const title = "Organization Settings";
+    const subtitle = "Manage your organization's settings.";
+
+    // Render as much of the page as we can in a loading state to avoid content shift
+    if (isLoading) {
+        return (
+            <div className="w-full">
+                <Header title={title} subtitle={subtitle} />
+                <div className="w-full">
+                    <SpinnerLoader />
+                </div>
+            </div>
+        );
+    }
+
+    if (!isOwner || !team) {
+        return <Redirect to={"/"} />;
```

Comment:
This didn't work for me. When I switch from org to my individual account I am not redirected.

Reply:
Thanks for catching this. I added a route-level check [here](https://github.com/gitpod-io/gitpod/pull/16277/commits/ba763ea09bb102a499353e8e8d738c48be5453a2) so we can redirect if there's no org selected right away. This simplified needing to check for an org in all of those page level components, and potentially kicking off api calls that will fail w/o one.